README.md

```diff
@@ -0,0 +1,304 @@
+
+        django load_exam docs/sample_questions.py
+
+	 Note that you can supply multiple Python files as arguments and all of
+	 those will be added to the database.
+    
+ 1. First run the python server provided. This ensures that the code is executed in a safe environment.  Do this like so:
+    
+        $ sudo python testapp/exam/code_server.py
+      
+    Put this in the background once it has started since this will not
+    return back the prompt.  It is important that the server be running
+    *before* students start attempting the exam.  Using sudo is
+    necessary since the server is run as the user "nobody".  This runs
+    on the ports configured in the settings.py file in the variable
+    "SERVER_PORTS".  The "SERVER_TIMEOUT" also can be changed there.
+    This is the maximum time allowed to execute the submitted code.
+    Note that this will likely spawn multiple processes as "nobody"
+    depending on the number of server ports specified.
+    
+ 1. The ```wsgi.py``` script should make it 
+ 	easy to deploy this using mod_wsgi.  You will need to add a line of the form:
+
+        WSGIScriptAlias / "/online_test/wsgi.py"
+
+	to your apache.conf.  For more details see the Django docs here:
+
+	https://docs.djangoproject.com/en/1.6/howto/deployment/wsgi/
+	
+ 1. Go to http://desired_host_or_ip:desired_port/admin
+
+ 1. Login with your credentials and look at the questions and modify if
+    needed.  Create a new Quiz, set the date and duration or
+    activate/deactivate the quiz.
+
+ 1. Now ask users to login at:
+
+        http://host:port/exam
+
+    And you should be all set.
+    
+ 1. Note that the directory "output" will contain directories, one for each
+    user.  Users can potentially write output into these that can be used
+    for checking later.
+
+ 1. As admin user you can visit http://host/exam/monitor  to view
+     results and user data interactively. You could also "grade" the
+     papers manually if needed.
+
+ 1. You may dump the results and user data using the results2csv and
+     dump_user_data commands.
+     
+ 1. The file docs/sample_questions.py is a template that you can use for your own questions.
+     
+ 1. Sometimes you might be in the situation where you are not hosted as
+    "host.org/exam/"  but as "host.org/foo/exam/" for whatever reason.  In
+    this case edit "settings.py" and set the "URL_ROOT"  to the root you
+    have to serve at.  In the above example for "host.org/foo/exam" set
+    URL_ROOT='/foo'.
+ 
+Development Settings
+====================
+
+To install this app follow the steps below:
+
+ 1. Clone this repository and cd to the cloned repo.
+  		```git clone  https://github.com/FOSSEE/online_test.git```
+ 
+ 1. Run:
+ 
+        django syncdb
+ 
+    Run:
+    
+        django migrate exam
+    
+ 1.  Add questions by editing the "docs/sample_questions.py" or any other file in the same format and then run the following:
```

Comment:
Formatting in this section is messed up.  View this file on github to see the issues.